### PR TITLE
chore(core, structure): various minor improvements for testing, addon dataset and search params

### PR DESCRIPTION
**Pull request by jordanl17**
### Description
This is part of a few PRs from the COREL feature branch which can be merged to next standalone. This PR includes smaller and miscellaneous changes.

### What to review
Broken down by commit:
1. [Creating a single addon dataset provider](https://github.com/sanity-io/sanity/pull/7386/commits/63de716b97ee69f563ec87b4078244b546a5950a)
2. [Additional context added to `TestProvider` for use when testing](https://github.com/sanity-io/sanity/pull/7386/commits/1d60623d2fb1afe061b93d08f8e8db19097cff25)
3. [Supporting stick search params in URLS](https://github.com/sanity-io/sanity/pull/7386/commits/b4487d2619ab5c6f57efc6b6115f6e08357b4d68)
4. [API to validate references in documents](https://github.com/sanity-io/sanity/pull/7386/commits/9f91858d2a159833f743c6f62ab42ac8950a2d15)

### Notes for release
N/A

### CODE REVIEW

**Review by RitaDias — COMMENTED** [review]
This looks good! I would prefer if someone else to review the code as well!
The only thing that comes to my mind is if we should add some basic tests to these things? Like the `validateDocumentWithReferences` (since it's a whole new file)

File: packages/sanity/src/core/validation/validateDocumentWithReferences.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+function shareLatestWithRefCount<T>() {
+  return shareReplay<T>({bufferSize: 1, refCount: true})
+}
+
+/**
+ * @internal
+ * Takes an observable of a document and validates it, including any references in the document.
+ * */
+export function validateDocumentWithReferences(
```
Comment (pedrobonamin) [inline packages/sanity/src/core/validation/validateDocumentWithReferences.ts:87]: Could we include the change to https://github.com/sanity-io/sanity/blob/corel/packages/sanity/src/core/store/_legacy/document/document-pair/validation.ts#L52-L53 in this pr?
This file is generated from there. It's extracting the validation logic from that file and making it reusable.

Comment (pedrobonamin) [reply packages/sanity/src/core/validation/validateDocumentWithReferences.ts:87]: We would need to move the copy the whole file from `corel` to here

Comment (jordanl17) [reply packages/sanity/src/core/validation/validateDocumentWithReferences.ts:87]: Yeh that one slipped through, I will add to this PR

**Pull request closed without merging**